hbw/app/controllers/hbw/tasks_controller.rb

```diff
@@ -27,15 +27,15 @@ def destroy
 
     def submit
       data = form_data.select { |key| fields_for_save.include?(key) }
+      file_fields.each do |file_field|
+        file_list_name = file_field['fileListName']
+        file_list = JSON.parse(data[file_list_name])
 
-      if files.present?
-        file_list = JSON.parse(data['homsOrderDataFileList'])
-
-        saved_files = minio_adapter.save_files(files)
+        saved_files = minio_adapter.save_files(file_field['files'])
 
         file_list += saved_files
 
-        data['homsOrderDataFileList'] = file_list.to_json
+        data[file_list_name] = file_list.to_json
```

Comment:
What will happen with existing form configurations without this new option? Ticket told about fallback.

Reply:
Done
Now if config of field doesn't have **file_list_name** option - we will write files to default bp variable **homsOrderDataFileList**

Fallback is in method **serialize** of **HBWFormFileUpload** component